### PR TITLE
Usage of HDF5 models for SVM classifiers

### PR DESCRIPTION
**Pull request by anuprulez**
This pull request creates an h5 ([HDF5](http://docs.h5py.org/en/stable/)) model of a trained SVM classifier. This model is shareable over Galaxy and can be reused to make predictions.

### CODE REVIEW

File: tools/sklearn/svm.xml [hunk 1]
```diff
@@ -29,12 +29,19 @@ with open(input_json_path, "r") as param_handler:
...
-    classifier_object = pickle.load(model_handler)
-
 header = 'infer' if params["selected_tasks"]["header"] else None
 data = pandas.read_csv("$selected_tasks.infile_data", sep='\t', header=header, index_col=None, parse_dates=True, encoding=None, tupleize_cols=False)
-prediction = classifier_object.predict(data)
+
+h5file = h5py.File("$infile_model", 'r')
+class_name = h5file.get("class_name").value
+class_path = h5file.get("class_path").value
+class_module = __import__(class_path, fromlist=[class_name])
```
[the remark this answers is not in the document]
Comment (qiagu) [reply tools/sklearn/svm.xml]: Do you need to apply a security check here?

File: tools/sklearn/svm.xml [hunk 2]
```diff
@@ -54,16 +61,26 @@ my_class = getattr(sklearn.svm, selected_algorithm)
 classifier_object = my_class(**options)
 classifier_object.fit(X, y)
 
-with open("$outfile_fit", 'wb') as out_handler:
-    pickle.dump(classifier_object, out_handler)
+classifier_dict = classifier_object.__dict__
+classifier_dict["class_path"] = classifier_object.__module__ 
+classifier_dict["class_name"] = classifier_object.__class__.__name__
```
[the remark this answers is not in the document]
Comment (qiagu) [reply tools/sklearn/svm.xml]: It seems most sklearn classes have "__module__" attribute. `pickle.whichmodule` or others could be an backup plan in case we meet problem.

**Comment by anuprulez** [comment]
@bgruening @qiagu In order to assign a unique name to each trained model (output ```h5``` dataset), I have renamed it by providing a label as ```label = tool_name.selected_algorithm```. With these names, a user can differentiate between different models. Do you think it makes more sense? 

outputmodelnames

File: tools/sklearn/svm.xml (hunk 1 above)
[the remark this answers is not in the document]
Comment (qiagu) [reply tools/sklearn/svm.xml]: I think we only have limited modules called in this tool. How about place all anticipated import in the very front?

[the remark this answers is not in the document]
Comment (qiagu) [reply tools/sklearn/svm.xml]: Also, how about call the SafePickler.find_class(module, name) to produce your class?

**Comment by qiagu** [comment]
The UI change looks good to me. Well done!

**Comment by bgruening** [comment]
We should also include the information about the model into a Galaxy datatype. We could do this by extending the H5 one https://github.com/galaxyproject/galaxy/blob/3c7eeecee52d5babf8039fd48ab161555086f84f/lib/galaxy/datatypes/binary.py#L695 or subclassing this and create a new one.

**Comment by bgruening** [comment]
Thanks!
